source/tutorial/gsfa/GSFA.rst

```diff
@@ -0,0 +1,414 @@
+
+4. `GSFA Optimization Problem`_
+
+5. `Linear GSFA Algorithm`_
+
+
+--------------
+
+.. _Classification using GSFA:
+
+1. Classification using GSFA 
+----------------------------
+
+To show the benefits of the model and its efficiency we will solve the
+classification task that is based on
+`‘breast_cancer’ <https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_breast_cancer.html>`__
+dataset of sklearn library. More precisely, we will use SVM Classifier
+as a baseline method and later see the effect of preprocessing data via
+GSFA.
+
+The breast cancer dataset is a classic and very easy binary
+classification dataset. Features are computed from a digitized image of
+a fine needle aspirate (FNA) of a breast mass. They describe
+characteristics of the cell nuclei present in the image. It can be found
+on `UCI Machine Learning
+Repository <https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+%28Diagnostic%29>`__
+
+**Number of instances:** 569
+
+**Number of attributes:** 30 numeric, predictive attributes and the
+class
+
+**Attribute information:**
+
+-  radius (mean of distances from center to points on the perimeter)
+-  texture (standard deviation of gray-scale values)
+-  perimeter
+-  area
+-  smoothness (local variation in radius lengths)
+-  compactness (perimeter^2 / area - 1.0)
+-  concavity (severity of concave portions of the contour)
+-  concave points (number of concave portions of the contour)
+-  symmetry
+-  fractal dimension (“coastline approximation” - 1)
+
+*The mean, standard error, and “worst” or largest (mean of the three
+worst/largest values) of these features were computed for each image,
+resulting in 30 features. For instance, field 0 is Mean Radius, field 10
+is Radius SE, field 20 is Worst Radius.*
+
+**Class:**
+
+- WDBC-Malignant
+- WDBC-Benign
+
+
+First let’s import the usual data science modules and library
+`mdp <https://mdp-toolkit.github.io>`__. We use it for GSFA
+implementation and other data processing tools.
+
+
+	>>> import matplotlib.pyplot as plt
+	>>> from sklearn import (datasets, model_selection, metrics)
+	>>> import numpy as np
+	>>> import mdp 
+	>>> from sklearn.svm import SVC
+
+
+Now from the sklearn let’s load the ‘breast_cancer’ dataset. We’ll use
+20% of the data for testing.
+
+
+	>>> data, label = datasets.load_breast_cancer(return_X_y=True
+	>>> data_train, data_test, label_train, label_test = \
+	... model_selection.train_test_split(data, label, test_size=0.2)
+
+We get the baseline classification quality.
+
+	>>> clf = SVC(gamma='auto')
+	>>> clf.fit(data_train, label_train)
+	>>> SVM_test = clf.predict(data_train)
+	>>> print("SVM test score: ", metrics.accuracy_score(label_train, SVM_test))
+
+
+**SVM train score:  1.0**
+
+
+	>>> clf = SVC(gamma='auto')
+	>>> clf.fit(data_train, label_train)
+	>>> SVM_test = clf.predict(data_test)
+	>>> print("SVM test score: ", metrics.accuracy_score(label_test, SVM_test))
+
+
+**SVM test score:  0.5614**
```

Comment:
I get 0.5877192982456141 running this code.

Reply:
You can add the result [like this](https://github.com/mdp-toolkit/mdp-docs/blame/master/source/examples/logmap/logmap.rst#L106) into the code.

Reply:
The result is random here. If you use the codesnippets extension it will add the following lines automatically to the generated source file. I am not sure whether this can or should be changed. In any case this fixes the seed and produces consistent results.

```
import mdp
import numpy as np
np.random.seed(0)
```